openstack/tools/func_test_tools/common.sh

```diff
@@ -44,3 +44,31 @@ apply_func_test_pr ()
         -f './src/test-requirements*.txt' \
         "$msg"
 }
+
+allocate_port ()
+{
+    # Returns address of port created.
+    #
+    local net_name=$1
+    local port_name=$2
+    openstack port create --network $net_name $port_name -c fixed-address -f value
+}
+
+create_zaza_vip ()
+{
+    # Allocates a vip ensuring to use existing ones if they exist.
+    #
+    # Returns the address of the vip.
+    #
+    local vip_id=$1
+    # We use the same naming convention as ../tools/allocate_vips.sh to avoid conflicts and re-use
+    # those vips.
+    vip_port_name=ps6-vip-ip$vip_id
+    vip_addr=$(openstack port show -c fixed_ips $vip_port_name -f yaml| yq .fixed_ips[0].ip_address)
+    if [[ $vip_addr = null ]]; then
```

Comment:
What if the ps6-vip-ip01 and ps6-vip-ip02 exist but are in use? I would need to review the generate_bundles script to make sure whether those 2 first IPs are likely to be chosen by it or not. It may look a bit complicated to coordinate sharing those VIPs with the ones pre-allocated by the ../tools/allocate_vips.sh script so it is best if we just make sure to adjust the range in the generate_bundles script to use the 18 last IPs, and keep ip01 and ip02 off its reach. That's all.

Reply:
this assumes they are not in use - we generally dont have quota for multiple deployments, its pretty unlikely that someone will have more than one HA deployment on the go at once. can fix later if needs be